### DISCUSSION TITLE
Long audio file, need help,.

### DISCUSSION THREAD

**Discussion by artyomboyko** [openai/whisper#726] [General]
Hello all. I'm new to audio and Whisper. Can anyone give me some pointers. How to process a long audio file in parts? Is there any mechanism that allows slicing audio in Python code (something like batches)?

As I understand it, this piece of code trims the audio to a length of 30 seconds:

``python
audio = whisper.pad_or_trim(audio)
```
How can I process a long audio file correctly?

**Comment by glangford** [comment]
I haven't had to do this myself, but you could identify periods of silence and split at those times, using for example ffmpeg directly or a library like pydub. Would that work for your audio? How long is it?

See for example 
https://askubuntu.com/questions/1264779/how-to-split-an-mp3-file-by-detecting-silent-parts

**Reply by glangford** [reply]
Within pydub, see eg.
https://github.com/jiaaro/pydub/blob/master/pydub/silence.py#L112

**Reply by artyomboyko** [reply]
[Solution](https://huggingface.co/blog/asr-chunking) found. 

```python
from transformers import pipeline

# Определим наличие графического ускорителя. Если доступен - используем.
pytorch_device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
print("Current PyTorch device is set to", pytorch_device)


audio_file_path = "/content/CS25 I Stanford Seminar - Transformers United: DL Models that have revolutionized NLP, CV, RL/audio.mp3"

pipe = pipeline(model="openai/whisper-small", device=pytorch_device)
predictions = pipe(files["audio_path"], chunk_length_s=20, stride_length_s=(5))

predictions = predictions["text"]
print(predictions)
```

**Reply by artyomboyko** [reply]
This option allows you to split the file when feeding it into the pipeline, during model inference.

**Reply by mattymatty** [reply]
> [Solution](https://huggingface.co/blog/asr-chunking) found.
> 
> ```python
> from transformers import pipeline
> 
> # Определим наличие графического ускорителя. Если доступен - используем.
> pytorch_device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
> print("Current PyTorch device is set to", pytorch_device)
> 
> 
> audio_file_path = "/content/CS25 I Stanford Seminar - Transformers United: DL Models that have revolutionized NLP, CV, RL/audio.mp3"
> 
> pipe = pipeline(model="openai/whisper-small", device=pytorch_device)
> predictions = pipe(files["audio_path"], chunk_length_s=20, stride_length_s=(5))
> 
> predictions = predictions["text"]
> print(predictions)
> ```

Where do you put this code?

**Reply by Funatiker** [reply]
Well. You put it in a Python file and execute it. However, it has two flaws:

1. The lines `audio_file_path = …` and `predictions = …` don't match. First the path to the audio file is defined, then something else is used. So you should replace `files["audio_path"]` by `audio_file_path`. Also you probably want to define a different path to an audio file.
2. `import torch` is missing

Besides that, the shown code converts an audio file to a text that is then printed in English. (If you want output in the original languages, set the `kwargs` for `language` to the original language and `task` to `transcribe` instead of `translate`. See https://huggingface.co/openai/whisper-large-v3 )

**Reply by artyomboyko** [reply]
@Funatiker Glad you figured out how to adapt someone else's code to solve your problem.

**Reply by Funatiker** [reply]
Yeah. Quite awesome that you posted your solution. So I could answer @mattymatty's question.
